src/app/core/entity-details/related-entities/related-entities.component.ts

```diff
@@ -64,7 +64,12 @@ export class RelatedEntitiesComponent<E extends Entity> implements OnInit {
     this.entityCtr = this.entities.get(this.entityType) as EntityConstructor<E>;
     this.isArray = isArrayProperty(this.entityCtr, this.property);
 
-    this.data = await this.entityMapper.loadType(this.entityType);
+    this.data = (await this.entityMapper.loadType<E>(this.entityType)).filter(
+      (e) =>
+        this.isArray
+          ? e[this.property].includes(this.entity.getId())
+          : e[this.property] === this.entity.getId(),
+    );
```

Comment:
Why do we need this now? Before this was covered through the filter below.

Reply:
The filter is passed as input and applied internally in entity-subrecord-component only, results do not change the `data` in the parent component (related-entities / related-entities-with-summary). The summary was therefore not limited to records related to the given entity.
This is not a perfect architecture here, filtering the same at two places ... but we may replace this by loading data from an index in the future. And it might also improve performance to pass a more limited initial data set to entity-subrecord